new-lamassu-admin/src/components/booleanPropertiesTable/BooleanPropertiesTable.js

```diff
@@ -16,102 +20,95 @@ const useStyles = makeStyles(booleanPropertiesTableStyles)
+                <div className={classes.transparentButton}>
+                  <button disabled={disabled} onClick={() => setEditing(true)}>
+                    {disabled ? <EditIconDisabled /> : <EditIcon />}
+                  </button>
+                </div>
+              )}
             </div>
-          )}
-        </div>
-        <Table className={classes.fillColumn}>
-          <TableBody className={classes.fillColumn}>
-            {radioGroupValues &&
-              radioGroupValues.map((element, idx) => (
-                <TableRow key={idx} size="sm" className={classes.tableRow}>
-                  <TableCell className={classes.leftTableCell}>
-                    {element.display}
-                  </TableCell>
-                  <TableCell className={classes.rightTableCell}>
-                    {editing && (
-                      <RadioGroup
-                        options={radioButtonOptions}
-                        value={element.value}
-                        onChange={event =>
-                          handleRadioButtons(
-                            element.name,
-                            event.target.value === 'true'
-                          )
-                        }
-                        className={classnames(
-                          classes.radioButtons,
-                          classes.rightTableCell
-                        )}
-                      />
-                    )}
-                    {!editing && (
-                      <BooleanCell
-                        className={classes.rightTableCell}
-                        value={element.value}
-                      />
-                    )}
-                  </TableCell>
-                </TableRow>
-              ))}
-          </TableBody>
-        </Table>
+            <PromptWhenDirty />
+            <Table className={classes.fillColumn}>
+              <TableBody className={classes.fillColumn}>
+                {elements.map((it, idx) => (
+                  <TableRow key={idx} size="sm" className={classes.tableRow}>
+                    <TableCell className={classes.leftTableCell}>
+                      {it.display}
+                    </TableCell>
+                    <TableCell className={classes.rightTableCell}>
+                      {editing && (
+                        <FormikField
+                          component={RadioGroup}
+                          name={it.name}
+                          options={radioButtonOptions}
+                          className={classnames(
+                            classes.radioButtons,
+                            classes.rightTableCell
+                          )}
+                        />
+                      )}
+                      {!editing && (
+                        <FormikField
+                          component={BooleanCell}
+                          name={it.name}
+                          className={classes.rightTableCell}
+                        />
```

Comment:
Move the BooleanCell component to within this file, also make it really simple like:

```js
(name) => {
  const { values } = useFormikContext()
  values[name] ? <TrueIcon/> : <FalseIcon/>
}
```

It doesn't need to be a formik field.